diff --git a/dool b/dool
index a70cb7a..9baf356 100755
--- a/dool
+++ b/dool
@@ -60,7 +60,7 @@ DOOL_DISKFILTER = re.compile(r'^([hsv]d[a-z]+\d+|cciss/c\d+d\d+p\d+|dm-\d+|md\d+
 class Options:
     def __init__(self, args):
         self.args          = args
-        self.bits          = True
+        self.bits          = False
         self.blackonwhite  = False
         self.count         = -1
         self.cpulist       = None
