diff --git a/gtweak/meson.build b/gtweak/meson.build
index 3d9a743..5c6d5c4 100644
--- a/gtweak/meson.build
+++ b/gtweak/meson.build
@@ -35,13 +35,11 @@ tweak_sources = [
   'tweaks/tweak_group_font.py',
   'tweaks/tweak_group_general.py',
   'tweaks/tweak_group_keymouse.py',
-  'tweaks/tweak_group_shell_extensions.py',
   'tweaks/tweak_group_startup.py',
   'tweaks/tweak_group_test.py',
   'tweaks/tweak_group_title_bar.py',
   'tweaks/tweak_group_top_bar.py',
   'tweaks/tweak_group_windows.py',
-  'tweaks/tweak_group_workspaces.py',
   'tweaks/tweak_group_xkb.py',
   'tweaks/tweak_wacom.py',
 ]
