diff --git a/cmake/FetchOnnxruntime.cmake b/cmake/FetchOnnxruntime.cmake
index eef5d67..0f0e723 100644
--- a/cmake/FetchOnnxruntime.cmake
+++ b/cmake/FetchOnnxruntime.cmake
@@ -96,40 +96,13 @@ elseif(OS_WINDOWS)
 
   install(IMPORTED_RUNTIME_ARTIFACTS Ort::DirectML DESTINATION "${OBS_PLUGIN_DESTINATION}")
 elseif(OS_LINUX)
-  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
-    if(USE_PREDEFINED_ONNXRUNTIME)
-      FetchContent_Declare(
-        Onnxruntime
-        URL "https://github.com/microsoft/onnxruntime/releases/download/v${Onnxruntime_VERSION}/onnxruntime-linux-aarch64-${Onnxruntime_VERSION}.tgz"
-        URL_HASH MD5=ebec0b185c9bec94fde884a97b144c04)
-    else()
-      FetchContent_Declare(
-        Onnxruntime
-        URL "${CUSTOM_ONNXRUNTIME_URL}"
-        URL_HASH MD5=${CUSTOM_ONNXRUNTIME_MD5})
-    endif()
-    FetchContent_MakeAvailable(Onnxruntime)
-    set(Onnxruntime_INSTALL_LIBS ${Onnxruntime_LINK_LIBS})
-  else()
-    if(USE_PREDEFINED_ONNXRUNTIME)
-      FetchContent_Declare(
-        Onnxruntime
-        URL "https://github.com/microsoft/onnxruntime/releases/download/v${Onnxruntime_VERSION}/onnxruntime-linux-x64-gpu-${Onnxruntime_VERSION}.tgz"
-        URL_HASH MD5=8d2f5ee9f449bdecb10a45715fe74c53)
-    else()
-      FetchContent_Declare(
-        Onnxruntime
-        URL "${CUSTOM_ONNXRUNTIME_URL}"
-        URL_HASH MD5=${CUSTOM_ONNXRUNTIME_MD5})
-    endif()
-    FetchContent_MakeAvailable(Onnxruntime)
-    set(Onnxruntime_INSTALL_LIBS
-        ${Onnxruntime_LINK_LIBS} "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime_providers_shared.so"
-        "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime_providers_cuda.so"
-        "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime_providers_tensorrt.so")
-  endif()
+  set(onnxruntime_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/../onnxruntime-linux-x64-gpu-${Onnxruntime_VERSION}")
   set(Onnxruntime_LINK_LIBS
       "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime.so.${Onnxruntime_VERSION}")
+  set(Onnxruntime_INSTALL_LIBS
+      ${Onnxruntime_LINK_LIBS} "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime_providers_shared.so"
+      "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime_providers_cuda.so"
+      "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime_providers_tensorrt.so")
   target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE ${Onnxruntime_LINK_LIBS})
   target_include_directories(${CMAKE_PROJECT_NAME} SYSTEM
                              PUBLIC "${onnxruntime_SOURCE_DIR}/include")
