diff --git a/cmake/FetchOnnxruntime.cmake b/cmake/FetchOnnxruntime.cmake
index f9231f6..0e19ff6 100644
--- a/cmake/FetchOnnxruntime.cmake
+++ b/cmake/FetchOnnxruntime.cmake
@@ -105,18 +105,7 @@ elseif(OS_LINUX)
     set(Onnxruntime_LINK_LIBS "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime.so.${Onnxruntime_VERSION}")
     set(Onnxruntime_INSTALL_LIBS ${Onnxruntime_LINK_LIBS})
   else()
-    if(USE_PREDEFINED_ONNXRUNTIME)
-      FetchContent_Declare(
-        Onnxruntime
-        URL "https://github.com/microsoft/onnxruntime/releases/download/v${Onnxruntime_VERSION}/onnxruntime-linux-x64-gpu-${Onnxruntime_VERSION}.tgz"
-        URL_HASH MD5=21b3eb42e0bee9e7b51cbf508352f188)
-    else()
-      FetchContent_Declare(
-        Onnxruntime
-        URL "${CUSTOM_ONNXRUNTIME_URL}"
-        URL_HASH MD5=${CUSTOM_ONNXRUNTIME_MD5})
-    endif()
-    FetchContent_MakeAvailable(Onnxruntime)
+    set(onnxruntime_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/../onnxruntime-linux-x64-gpu-${Onnxruntime_VERSION}")
     set(Onnxruntime_LINK_LIBS "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime.so.${Onnxruntime_VERSION}")
     set(Onnxruntime_INSTALL_LIBS
         ${Onnxruntime_LINK_LIBS} "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime_providers_shared.so"
