diff --git a/cmake/FetchOnnxruntime.cmake b/cmake/FetchOnnxruntime.cmake
index 3bf877f..0958ea2 100644
--- a/cmake/FetchOnnxruntime.cmake
+++ b/cmake/FetchOnnxruntime.cmake
@@ -113,18 +113,7 @@ elseif(OS_LINUX)
         "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime.so.${Onnxruntime_VERSION}")
     set(Onnxruntime_INSTALL_LIBS ${Onnxruntime_LINK_LIBS})
   else()
-    if(USE_PREDEFINED_ONNXRUNTIME)
-      FetchContent_Declare(
-        Onnxruntime
-        URL "https://github.com/microsoft/onnxruntime/releases/download/v${Onnxruntime_VERSION}/onnxruntime-linux-x64-gpu-${Onnxruntime_VERSION}.tgz"
-        URL_HASH MD5=8d2f5ee9f449bdecb10a45715fe74c53)
-    else()
-      FetchContent_Declare(
-        Onnxruntime
-        URL "${CUSTOM_ONNXRUNTIME_URL}"
-        URL_HASH MD5=${CUSTOM_ONNXRUNTIME_MD5})
-    endif()
-    FetchContent_MakeAvailable(Onnxruntime)
+    set(onnxruntime_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/../onnxruntime-linux-x64-gpu-${Onnxruntime_VERSION}")
     set(Onnxruntime_LINK_LIBS
         "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime.so.${Onnxruntime_VERSION}")
     set(Onnxruntime_INSTALL_LIBS
