diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9af53c4..6d163b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,9 +74,6 @@ if(APPLE)
 elseif(MSVC)
   add_subdirectory(src/update-checker/WinRTHttpClient)
   target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE WinRTHttpClient)
-elseif(UNIX)
-  add_subdirectory(src/update-checker/CurlClient)
-  target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE CurlClient)
 endif()
 
 target_sources(
@@ -85,9 +82,6 @@ target_sources(
           src/ort-utils/ort-session-utils.cpp
           src/obs-utils/obs-utils.cpp
           src/obs-utils/obs-config-utils.cpp
-          src/update-checker/github-utils.cpp
-          src/update-checker/update-checker.cpp
-          src/update-checker/UpdateDialog.cpp
           src/background-filter-info.c
           src/background-filter.cpp
           src/enhance-filter.cpp
diff --git a/src/plugin-main.c b/src/plugin-main.c
index 8bc2a61..d4eb579 100644
--- a/src/plugin-main.c
+++ b/src/plugin-main.c
@@ -20,8 +20,6 @@ with this program. If not, see <https://www.gnu.org/licenses/>
 
 #include "plugin-support.h"
 
-#include "update-checker/update-checker.h"
-
 OBS_DECLARE_MODULE()
 OBS_MODULE_USE_DEFAULT_LOCALE(PLUGIN_NAME, "en-US")
 
@@ -40,8 +38,6 @@ bool obs_module_load(void)
 	obs_log(LOG_INFO, "Plugin loaded successfully (version %s)",
 		PLUGIN_VERSION);
 
-	check_update();
-
 	return true;
 }
 
